compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1629,18 +1629,30 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     infer::LateBoundRegionConversionTime::HigherRankedType,
                     bound_predicate.rebind(data),
                 );
-                let normalized_ty = ocx.normalize(
-                    &obligation.cause,
-                    obligation.param_env,
-                    self.tcx
-                        .mk_projection(data.projection_ty.item_def_id, data.projection_ty.substs),
-                );
+                let unnormalized_term = match data.term.unpack() {
+                    ty::TermKind::Ty(_) => self
+                        .tcx
+                        .mk_projection(data.projection_ty.item_def_id, data.projection_ty.substs)
+                        .into(),
+                    ty::TermKind::Const(ct) => self
+                        .tcx
+                        .mk_const(
+                            ty::UnevaluatedConst {
+                                def: ty::WithOptConstParam::unknown(data.projection_ty.item_def_id),
+                                substs: data.projection_ty.substs,
+                            },
+                            ct.ty(),
```

Comment:
This is the type from the expected const, but we don't carry the type from the LHS of the projection equality predicate. Anyways, not a huge issue, since we always expect these types to be equal (we assert for it in other places, I'm pretty sure).